students/ejackie/session04/trigrams_dict.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+
+##################################################
+## Author: Jackie Cheung
+## Date: 2018/10/30
+## Version: 0.1
+## Class: Au2018-Py210B
+## Description: Trigrams – Simple Text Manipulation (Kata Fourteen: Tom Swift Under Milk Wood)
+##################################################
+
+import sys
+import random
+
+#words = "I wish I may I wish I might".split()
+
+def read_in_data(filename):
+    """
+    read text from specified file
+
+    """ 
+    with open(filename, 'r') as f:
+        in_data = f.read()
+    f.closed
+    True
+    print('in_data: \n', in_data)
+    return in_data
+
+
+def make_words(in_data):
+    """
+    clean up the special character from the text and split the words properly
+
+    """ 
+    words = in_data.translate(in_data.maketrans('', '', '^[!@#$%^&*)(+=.,_-]/'))
+    print('words: \n', words)
+    return words.split()
+
+
+def build_trigrams(words):
+    """
+    build up the trigrams dict from the list of words
+
+    returns a dict with:
+       keys: word pairs
+       values: list of followers
+    """
+    trigrams = {}
+
+    # build up the dict here!
+
+    for i in range(len(words)-2):
+        pair = words[i:i + 2]
+        follower = words[i + 2]
+        if tuple(pair) in trigrams:
+            trigrams[tuple(pair)] = [trigrams[tuple(pair)], follower]
```

Comment:
Minor: Why not `trigrams[tuple(pair)].append(follower)`? Probably this has something to do with your line 52, where you assign `follower` as the value of `trigrams[tuple(pair)]`, not `[follower]`. You could also use `trigrams = defaultdict(list)` in line 47, after adding `from collections import defaultdict` at the top. Then there's even no need to check line 54 at all.